pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,523 @@
+
+    Args:
+        qnode (pennylane.QNode): Quantum node to be reconstructed, representing a
+            circuit that outputs an expectation value.
+        ids (dict or Sequence or str): Indices for the QNode parameters with respect to which
+            the QNode should be reconstructed as a univariate function, per QNode argument.
+            Each key of the dict, entry of the list, or the single ``str`` has to be the name
+            of an argument of ``qnode`` .
+            If a ``dict`` , the values of ``ids`` have to contain the parameter indices
+            for the respective array-valued QNode argument represented by the key.
+            If a ``list`` , the parameter indices are inferred from ``nums_frequency`` if
+            given or ``spectra`` else.
+            If ``None``, all keys present in ``nums_frequency`` / ``spectra`` are considered.
+        nums_frequency (dict[dict]): Numbers of integer frequencies -- and biggest
+            frequency -- per QNode parameter. The keys have to be argument names of ``qnode``
+            and the inner dictionaries have to be mappings from parameter indices to the
+            respective integer number of frequencies. If the QNode frequencies are not contiguous
+            integers, the argument ``spectra`` should be used to save evaluations of ``qnode`` .
+            Takes precedence over ``spectra`` and leads to usage of equidistant shifts.
+        spectra (dict[dict]): Frequency spectra per QNode parameter.
+            The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+            be mappings from parameter indices to the respective frequency spectrum for that
+            parameter. Ignored if ``nums_frequency!=None``.
+        shifts (dict[dict]): Shift angles for the reconstruction per QNode parameter.
+            The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+            be mappings from parameter indices to the respective shift angles to be used for that
+            parameter. For :math:`R` non-zero frequencies, there must be :math:`2R+1` shifts
+            given. Ignored if ``nums_frequency!=None``.
+
+    Returns:
+        function: Function which accepts the same arguments as the QNode.
+            When called, this function will return a dictionary of dictionaries,
+            formatted like ``nums_frequency`` or ``spectra`` ,
+            that contains the univariate reconstructions per QNode parameter.
+
+    For each provided ``id`` in ``ids``, the QNode is restricted to varying the single QNode
+    parameter corresponding to the ``id`` . This univariate function is then reconstructed
+    via a Fourier transform or Dirichlet kernels, depending on the provided input.
+    Either the ``spectra`` of the QNode with respect to its input parameter or the numbers
+    of frequencies, ``nums_frequency`` , per parameter must be provided.
+
+    **Usage Details**
```

Comment:
```suggestion
    **Details**
```

I recommend shortening this to a 1-2 paragraphs if you can, and moving the bulk of the details into a `.. UsageDetails::` section 🙂 

At least for me, I had to scroll down first to read the example to get a good understanding of how everything works together; visually, it is much more instructive.

I actually think how you did it in [`qnode_spectrum`](https://pennylane--1864.org.readthedocs.build/en/1864/code/api/pennylane.fourier.qnode_spectrum.html) is perfect! 

- Short summary above the arguments.

- *Details*: After the arguments. A short paragraph with the basic maths and links to papers.

- *Example*: a minimal example required to visually describe the logic, with minimal text explanation.

- *Usage details*: a collapsed by default section with more complicated examples, various different use-cases, and a description of all gotchas.

Reply:
I will not repeat the basic math part in this one, I think. It is going to be contained in the Rotosolve docstring, there are 2 demos covering it and the Fourier model introduces the concepts as well :sweat_smile: 
But I agree, this is a much nicer structure! :)

Reply:
:+1: